espresso/environment/2_espresso_liveness_test.go

```diff
@@ -119,3 +131,285 @@ func TestE2eDevNetWithEspressoEspressoDegradedLiveness(t *testing.T) {
+
+		// Start consuming Batches from the Streamer
+		// We cannot guarantee the that we will receive only the batches that
+		// correspond to the transactions we submitted, so we will need to
+		// keep track of the batches we receive and match them up with the
+		// transactions we submitted.
+		//
+		// Luckily, it seems that the Block contained within the batch will
+		// maintain the same block hash, and the transaction hashes will match
+		// for the transactions beyond the first in the block.
+		wg.Add(1)
+		go (func(ctx context.Context, wg *sync.WaitGroup, streamer espresso.EspressoStreamer[derive.EspressoBatch]) {
+			cfg := system.RollupConfig
+			defer wg.Done()
+			for {
+				select {
+				default:
+				case <-ctx.Done():
+					// We are being told to exit, so we exit
+					return
+				}
+
+				if !streamer.HasNext(ctx) {
+					if err := streamer.Update(ctx); err != nil {
+						// Try again after a short delay if we we fail to
+						// update the streamer.
+						time.Sleep(50 * time.Millisecond)
+					}
+					continue
+				}
+
+				// consume all of the available batches
+				for batch := streamer.Next(ctx); batch != nil; batch = streamer.Next(ctx) {
+					block, err := batch.ToBlock(cfg)
+					if have, want := err, error(nil); have != want {
+						return
+					}
+
+					espressoReceipts[block.Hash()] = espressoReceived{
+						batch:    batch,
+						block:    block,
+						received: time.Now(),
+					}
+				}
+			}
+		})(streamBlocksCtx, &wg, streamer)
+	}
+
+	type submission struct {
+		receipt   *geth_types.Receipt
+		created   time.Time
+		submitted time.Time
+		received  time.Time
+	}
+	var submissions []submission
+
+	// The number of transaction we want to submit to the L2.
+	// This will also correspond to the number of batches we expect to receive
+	// from the Espresso streamer.
+	const N = 10
+	{
+		for i := 0; i < N; i++ {
+			// Create teh transaction
+			tx := geth_types.MustSignNewTx(system.Cfg.Secrets.Bob, geth_types.LatestSignerForChainID(system.Cfg.L2ChainIDBig()), &geth_types.DynamicFeeTx{
+				ChainID:   system.Cfg.L2ChainIDBig(),
+				Nonce:     uint64(i),
+				To:        &addressAlice,
+				Value:     big.NewInt(1),
+				GasTipCap: big.NewInt(10),
+				GasFeeCap: big.NewInt(200),
+				Gas:       21_000,
+			})
+			created := time.Now()
+
+			// Send the transaction
+			err := l2Seq.SendTransaction(ctx, tx)
+			if have, want := err, error(nil); have != want {
+				t.Fatalf("Sending L2 tx:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+			}
+
+			// We have submitted the transaction to the L2, successfully.
+			submitted := time.Now()
+
+			// Wait for the receive
+			receipt, err := wait.ForReceiptOK(ctx, l2Seq, tx.Hash())
+			if have, want := err, error(nil); have != want {
+				t.Fatalf("Waiting for L2 tx:\nhave:\n\t\"%v\"\nwant:\n\t\"%v\"\n", have, want)
+			}
+
+			// We not have a receipt from the L2 Sequencer, indicating that
```

Comment:
```suggestion
			// We now have a receipt from the L2 Sequencer, indicating that
```

Reply:
Addressed in cdf5e56d23f7ae4b5ff688378e345e6d856e162f.